src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -137,7 +137,7 @@ public ParticipationResource(ParticipationService participationService, Programm
         log.debug("REST request to start Exercise : {}", exerciseId);
         Exercise exercise = exerciseService.findOne(exerciseId);
         Course course = exercise.getCourse();
-        if (!courseService.userHasAtLeastStudentPermissions(course)) {
+        if (!authCheckService.isAtLeastStudentInCourse(course, null)) {
```

Comment:
I stumbled upon the use of the deprecated method `userHasAtLeastStudentPermission(course)` when analyzing this issue: https://sentry.io/share/issue/4c27fa7395c74317ba800d75eaf7b35d/

The error was only thrown 980 times, because the participation was triggered on the prod environment with 1000 test users and only 20 had the right permissions.

Reply:
Sentry link is now outdated :(